### ISSUE TITLE
Test: fault tolerance at master side

### ISSUE THREAD

**Issue by ttyusupov** [yugabyte/yugabyte-db#8645]
From https://phabricator.dev.yugabyte.com/D8253#243515:
"
We should also consider some unhappy paths:
- crash or leader failover before sending out the split tasks
- long enough partition while trying to send out the splits so that they timeout and not get executed

We'll have some tablets in CREATING state, which have their table_id set, but the split does not get retried. For a timeout, they're also not added to the TableInfo object at all. For the failover case, I believe they'll automatically be added to the tablet, using AddTablet (in the tablet loader code).
"

**Comment by bmatican** [comment]
@SrivastavaAnubhav can you chime in if we have coverage by now for all of these?

**Comment by SrivastavaAnubhav** [comment]
These should be handled by automatic tablet splitting, if it's enabled (but it's not enabled by default yet).

**Comment by bmatican** [comment]
@SrivastavaAnubhav oh, are you saying that if automatic is turned off, but a user does manual splits, we might not be re-triggering splits after master failover, if neeed? If so, might be worth discussing in next week's sync, as that seems a bit off.

**Comment by SrivastavaAnubhav** [comment]
Yeah, we should have some common method that gets called on restart.
